cunumeric/module.py

```diff
@@ -2448,6 +2448,113 @@ def flip(m: ndarray, axis: Optional[NdShapeLike] = None) -> ndarray:
 # Generating index arrays
 
 
+@add_boilerplate("arr", "mask", "vals")
+def place(arr: ndarray, mask: ndarray, vals: ndarray) -> None:
+    """
+    Change elements of an array based on conditional and input values.
+
+    Parameters
+    ----------
+    arr : array_like
+        Array to put data into.
+    mask : array_like
+        Boolean mask array. Must have the same size as `arr`.
+    vals : 1-D sequence
+        Values to put into `arr`. Only the first N elements are used,
+        where N is the number of True values in mask. If vals is smaller
+        than N, it will be repeated, and if elements of a are to be masked,
+        this sequence must be non-empty.
+
+    See Also
+    --------
+    numpy.copyto, numpy.put, numpy.take, numpy.extract
+
+    Availability
+    --------
+    Multiple GPUs, Multiple CPUs
+    """
+    if mask.size != arr.size:
+        raise ValueError("arr array and condition array must be of same size")
+
+    if vals.ndim != 1:
+        raise ValueError("vals array has to be 1-dimensional")
+
+    if mask.shape != arr.shape:
+        mask_reshape = reshape(mask, arr.shape)
+    else:
+        mask_reshape = mask
+
+    num_values = count_nonzero(mask_reshape)
+    if num_values == 0:
+        return
+
+    if vals.size == 0:
+        raise ValueError("vals array cannot be empty")
+
+    if num_values > vals.size:
+        # repeat vals to achieve same length as num_values
+        vals_resized = repeat(
+            reshape(vals, (1, vals.size)),
+            math.ceil(num_values / vals.size),
+            axis=0,
+        ).flatten()[:num_values]
+
+    elif num_values < vals.size:
+        vals_resized = vals[:num_values]
+    else:
+        vals_resized = vals
+
+    if mask_reshape.dtype == bool:
+        arr._thunk.set_item(mask_reshape._thunk, vals_resized._thunk)
+    else:
+        bool_mask = mask_reshape.astype(bool)
```

Comment:
shouldn't mask be always bool?
From numpy documentation:
```
mask : array_like
Boolean mask array. Must have the same size as a.
```

Reply:
I tested that numpy does work with other types as well, just like with extract. Should I remove support?

Reply:
No, we probably want to have the same support as NumPy.